table/column.go

```diff
@@ -285,7 +285,7 @@ func CheckOnce(cols []*Column) error {
 
 // CheckNotNull checks if nil value set to a column with NotNull flag is set.
 func (c *Column) CheckNotNull(data types.Datum) error {
-	if mysql.HasNotNullFlag(c.Flag) && data.IsNull() {
+	if mysql.HasNotNullFlag(c.Flag) && data.IsNull() || mysql.HasPreventNullInsertFlag(c.Flag) && data.IsNull() {
```

Comment:
Please use bracket to wrap `mysql.HasNotNullFlag(c.Flag) && data.IsNull()` and `mysql.HasPreventNullInsertFlag(c.Flag) && data.IsNull()`